pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/merge_rollup/MergeRollupTaskUtils.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.pinot.pql.parsers.utils.Pair;
+import org.apache.pinot.spi.utils.TimeUtils;
+
+
+public class MergeRollupTaskUtils {
+
+  public static class MergeProperties {
+    private final long _bufferMs;
+    private final long _maxNumRecordsPerSegment;
+    private final long _maxNumRecordsPerTask;
+
+    public MergeProperties(long bufferMs, long maxNumRecordsPerSegment, long maxNumRecordsPerTask) {
+      _bufferMs = bufferMs;
+      _maxNumRecordsPerSegment = maxNumRecordsPerSegment;
+      _maxNumRecordsPerTask = maxNumRecordsPerTask;
+    }
+
+    public long getBufferMs() {
+      return _bufferMs;
+    }
+
+    public long getMaxNumRecordsPerSegment() {
+      return _maxNumRecordsPerSegment;
+    }
+
+    public long getMaxNumRecordsPerTask() {
+      return _maxNumRecordsPerTask;
+    }
+  }
+
+  private static final String[] validMergeProperties = {
+      MinionConstants.MergeRollupTask.BUFFER_TIME,
+      MinionConstants.MergeRollupTask.MAX_NUM_RECORDS_PER_SEGMENT,
+      MinionConstants.MergeRollupTask.MAX_NUM_RECORDS_PER_TASK
+  };
+
+  public static Map<String, ValueAggregatorFactory.ValueAggregatorType> getRollupAggregationTypeMap(Map<String, String> mergeRollupConfig) {
+    Map<String, ValueAggregatorFactory.ValueAggregatorType> rollupAggregationTypeMap = new HashMap<>();
+    for (Map.Entry<String, String> entry : mergeRollupConfig.entrySet()) {
+      if (entry.getKey().startsWith(MinionConstants.MergeRollupTask.AGGREGATE_KEY_PREFIX)) {
+        rollupAggregationTypeMap.put(getAggregateColumn(entry.getKey()),
+            ValueAggregatorFactory.ValueAggregatorType.valueOf(entry.getValue().toUpperCase()));
+      }
+    }
+    return rollupAggregationTypeMap;
+  }
+
+  public static Map<MergeRollupTaskMetadata.Granularity, MergeProperties> getAllMergeProperties(Map<String, String> mergeRollupConfig) {
+    Map<MergeRollupTaskMetadata.Granularity, Map<String, String>> mergePropertiesMap = new HashMap<>();
+    for (Map.Entry<String, String> entry : mergeRollupConfig.entrySet()) {
+      if (entry.getKey().startsWith(MinionConstants.MergeRollupTask.MERGE_KEY_PREFIX)) {
+        Pair<MergeRollupTaskMetadata.Granularity, String> pair = getGranularityAndPropertyPair(entry.getKey());
+        MergeRollupTaskMetadata.Granularity granularity = pair.getFirst();
+        String mergeProperty = pair.getSecond();
+        mergePropertiesMap.putIfAbsent(granularity, new HashMap<>());
+        mergePropertiesMap.get(granularity).put(mergeProperty, entry.getValue());
+      }
+    }
+
+    Map<MergeRollupTaskMetadata.Granularity, MergeProperties> allMergeProperties = new HashMap<>();
+    for (Map.Entry<MergeRollupTaskMetadata.Granularity, Map<String, String>> entry : mergePropertiesMap.entrySet()) {
+      Map<String, String> properties = entry.getValue();
+      MergeProperties mergeProperties = new MergeProperties(
+          TimeUtils.convertPeriodToMillis(properties.get(MinionConstants.MergeRollupTask.BUFFER_TIME)),
```

Comment:
We also need to cover `mergeType` for each gran

Reply:
Added